### ISSUE TITLE
Track source locations of preconditions

### ISSUE THREAD

**Issue by langston-barrett** [GaloisInc/saw-script#415]
While implementing #414, I noticed that `matchArg` can only report source locations on the level of overrides. It would be nice if we could provide the user with line numbers for each precondition. 

This would probably involve changing
```haskell
  , _csArgBindings     :: Map Integer (t, SetupValue) -- ^ function arguments
```
to something like
```haskell
  , _csArgBindings     :: Map Integer (ArgSpec t)
```
where
```haskell
data ArgSpec t = ArgSpec { _val :: SetupValue, _loc :: What4.ProgramLoc, _type :: t }
```

**Comment by siddharthist** [comment]
This is especially important for points-to assertions. I might have tens of these in my preconditions, many with the same types involved. It's quite difficult to tell which one should be fixed when something goes wrong, especially if they're all symbolic.

**Comment by siddharthist** [comment]
It looks like the problem is that `crucible_precond`, `crucible_points_to`, etc. use `getPosition` from the `TopLevel` monad, which ignores their position inside the `CrucibleSetup` block.

**Comment by siddharthist** [comment]
The case of `crucible_precond`, `crucible_points_to`, etc. was fixed in #477. `matchArg` could still use the solution outlined above.

**Comment by robdockins** [comment]
PR #1679 implements most of this, to the point I think it's time to close this ticket.
